### PR TITLE
chore: change changelog creation to leverage GH releases

### PR DESCRIPTION
**Pull request by gruebel**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally we encourage adding a scope to the prefix, which indicates the targeted framework, otherwise 'general' will be assumed.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

- changed the way on how the changelog is created, additionallythe GH release will now also get the release notes for noteworthy changes
- I hope I chose the right token 🤞  

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ayajbara — APPROVED** [review]

File: .github/pr-title-checker-config.json [hunk 1]
```diff
@@ -5,18 +5,13 @@
...
-      "fix: ",
-      "feat: ",
-      "break: ",
-      "docs: ",
-      "chore: ",
-      "platform: "
+      "chore: "
     ],
-    "regexp": "(fix|feat|break|docs|chore|platform)\\((arm|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph)\\): "
+    "regexp": "(fix|feat|break|docs|chore|platform)\\((argo|arm|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph)\\): "
```
Comment (Eliran-Turgeman) [inline .github/pr-title-checker-config.json:10]: Should we have a 1:1 map between frameworks to the regex?
If so, `github_actions` is missing (or should we use `github` instead?)

**Review by nimrodkor — COMMENTED** [review]
@gruebel Need to be in the release env to fetch that secret...

File: .github/release-changelog-config.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            "target": "$1"
+        }
+    ],
+    "transformers": [
+        {
+            "pattern": "([^\\(]+)\\(?([^\\)]+)?\\)?: (.+)",
+            "target": "- **$2:** $3"
+        }
+    ],
+    "max_pull_requests": 100,
```
Comment (Eliran-Turgeman) [inline .github/release-changelog-config.json:44]: Probably not so important but just curious - isn't every release created after each PR is merged?
What's the purpose of this limit? 🤔

**Review by Saarett — APPROVED** [review]

File: .github/release-changelog-config.json (hunk 2 above)
Comment (gruebel) [reply .github/release-changelog-config.json:44]: nono, the GitHub release is only created once per day at night. So, I chose just a number, which is high enough, but not to crazy 😄

File: .github/pr-title-checker-config.json (hunk 1 above)
Comment (gruebel) [reply .github/pr-title-checker-config.json:10]: I wanted it to keep a bit more simple for now, but if we feel like we want to separate it more, then sure lets do it in a later step 🙂

**Comment by gruebel** [comment]
> @gruebel Need to be in the release env to fetch that secret...

since it runs only on the `master` branch it should use the env automatically

**Pull request merged**